pulsar/impl_message.go

```diff
@@ -118,6 +120,24 @@ func newMessageID(ledgerID int64, entryID int64, batchIdx int, partitionIdx int)
 	}
 }
 
+// creates a new messageID object from the given MessageID, consumers is the amount of
+// consumers, 1+ for Consumer, 1 for Reader.
+func newMessageIDFromInterface(msgID MessageID, consumers int) (*messageID, error) {
```

Comment:
This is not returning a new message id it's the same struct right?

Reply:
msgID is an interface, the returned value the internal struct

Reply:
The method name is confusing to me something with new should return a new object. This is just casting it to something.

Reply:
I renamed and moved the function so that it later can be used by `Reader` as well. This function existed before, it just moved. It converts the given interface implementation to the internal type. If this conversion fails, the callers like NackID can not succeed and should return an error to their caller. I propose this change instead of logging and not returning an error. This would be an API change tho.